### PR TITLE
Implemented priv_debugGetStateRoot JSON RPC API.

### LINKED ISSUE #764: Implement priv_debugGetStateRoot JSON RPC API
As a privacy developer, I want to be able to retrieve the state root of a privacy group for a specified block hash so that I can confirm the state root matches on different nodes when debugging. 

Parameters - privacy group ID and block hash 
Result - state root of the privacy group for the specified block hash 

If the privacy group ID specified doesn't exist or node requesting does not have access, return the same error message as find privacy group in these cases.

### PR DESCRIPTION
**Pull request by mark-terry**
Signed-off-by: Mark Terry <mark.terry@consensys.net>

## PR description
Implements a "priv_debugGetStateRoot" JSON RPC API that allows querying the state root.

## Fixed Issue(s)
Fixes #764 

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PrivJsonRpcMethods.java [hunk 1]
```diff
@@ -65,57 +66,44 @@ protected RpcApi getApiGroup() {
...
-              getBlockchainQueries(),
-              getPrivacyQueries(),
-              privacyController,
-              enclavePublicKeyProvider),
-          new PrivNewFilter(filterManager, privacyController, enclavePublicKeyProvider),
-          new PrivUninstallFilter(filterManager, privacyController, enclavePublicKeyProvider),
-          new PrivGetFilterLogs(filterManager, privacyController, enclavePublicKeyProvider),
-          new PrivGetFilterChanges(filterManager, privacyController, enclavePublicKeyProvider));
+
+    final Map<String, JsonRpcMethod> RPC_METHODS =
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PrivJsonRpcMethods.java:70]: I am happy to see this refactor

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PrivJsonRpcMethods.java:70]: adding a unit test would take it to next level

**Review by macfarla — APPROVED** [review]
LGTM

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PrivJsonRpcMethods.java (hunk 1 above)
Comment (mark-terry) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PrivJsonRpcMethods.java:70]: The existing tests have good coverage of what I've changed. 👍

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivacyController.java [hunk 2]
```diff
@@ -100,5 +100,8 @@ void verifyPrivacyGroupContainsEnclavePublicKey(
 
   PrivateTransactionSimulator getTransactionSimulator();
 
-  Optional<Hash> getBlockHashByBlockNumber(Optional<Long> blockNumber);
+  Optional<Hash> getBlockHashByBlockNumber(final Long blockNumber);
```
Comment (lucassaldanha) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivacyController.java:103]: Do we need to expose this method in this interface? Isn't it used internally?

I recon we could just find the blockhash from the number inside `getStateRootByBlockNumber()` method.

**Review by lucassaldanha — APPROVED** [review]
LGTM. Only thing it needs is a CHANGELOG entry :)

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PrivJsonRpcMethods.java (hunk 1 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PrivJsonRpcMethods.java:70]: even better

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,12 +1,16 @@
 # Changelog
 
+### Additions and Improvements
+
+* Added `priv_debugGetStateRoot` JSON RPC API to retrieve the state root of a specified privacy group. [\#1326](https://github.com/hyperledger/besu/pull/1326)
```
Comment (macfarla) [inline CHANGELOG.md:5]: s/JSON RPC/JSON-RPC

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+
 ## 1.5.3 
 
 ### Additions and Improvements
 
 * The EvmTool now processes State Tests from the Ethereum Reference Tests. [\#1311](https://github.com/hyperledger/besu/pull/1311)
 * Experimental dns support added via the `Xdns-enabled` and `Xdns-update-enabled` CLI commands. [\#1247](https://github.com/hyperledger/besu/pull/1247)
-* Add genesis config option `ecip1017EraRounds` for Ethereum Classic chanis. [\#1329](https://github.com/hyperledger/besu/pull/1329)
+* Add genesis config option `ecip1017EraRounds` for Ethereum Classic chains. [\#1329](https://github.com/hyperledger/besu/pull/1329)
```
Comment (macfarla) [inline CHANGELOG.md:13]: good spot

**Pull request merged**